polaris-core/src/main/java/org/apache/polaris/core/config/PolarisConfigurationStore.java

```diff
@@ -99,7 +99,27 @@ public interface PolarisConfigurationStore {
    */
   default <T> @Nonnull T getConfiguration(
       @Nonnull RealmContext realmContext, PolarisConfiguration<T> config) {
-    T result = getConfiguration(realmContext, config.key, config.defaultValue);
+    T result = getConfiguration(realmContext, config.key);
+
+    if (result == null && config.hasLegacyKeys()) {
+      for (String legacyKey : config.legacyKeys()) {
```

Comment:
This will lead to nondeterministic behavior if two legacy keys for a feature have different values